Src/library/net/network/socket/network_socket.e

```diff
@@ -209,32 +234,63 @@ feature -- Status setting
 		require
 			non_negative: n >= 0
 		do
-			if n > 0 then
-				timeout := n
-			else
-				timeout := default_timeout
-			end
+			set_fine_timeout (n.to_natural_64)
 		ensure
 			timeout_set: timeout = n or timeout = default_timeout
 		end
 
+	set_fine_timeout (nb_nano_seconds: NATURAL_64)
```

Comment:
Add precondition to show that the argument is limited as the underlying C API only accepts 32-bit integers for seconds